plugin/trino-opensearch/src/main/java/io/trino/plugin/opensearch/OpenSearchColumnHandle.java

```diff
@@ -13,28 +13,50 @@
  */
 package io.trino.plugin.opensearch;
 
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableList;
 import io.trino.spi.connector.ColumnHandle;
 import io.trino.spi.type.Type;
 
+import java.util.List;
+
 import static java.util.Objects.requireNonNull;
 
 public record OpenSearchColumnHandle(
-        String name,
+        List<String> path,
         Type type,
         DecoderDescriptor decoderDescriptor,
-        boolean supportsPredicates)
+        boolean supportsPredicates,
+        String qualifiedName)
```

Comment:
What happens if someone calls the constructor with the unexpected `List<String> path` & `String qualifiedName`?

By the way, `qualifiedName` might be a little misleading. Some people may think that the name contains `catalog.schema.table`.

Reply:
Good point, I have added a check and renamed it to dereferencePath.